code/modules/library/lib_machines.dm

```diff
@@ -374,6 +377,18 @@
 			var/obj/machinery/libraryscanner/scan = get_scanner()
 			data["has_scanner"] = !!(scan)
 			data["has_cache"] = !!(scan?.cache)
+
+			var/cooldown_number = (COOLDOWN_TIMELEFT(src, newscaster_cooldown))
+			var/active_newscaster_cooldown = !!(COOLDOWN_FINISHED(src, newscaster_cooldown))
```

Comment:
why the typecast?

Reply:
i thought there was some esoteric tgui/js reason to do this whoops